chevah/compat/testing/testcase.py

```diff
@@ -1020,6 +1021,20 @@ def callCleanup(self):
 
         self.__cleanup__ = []
 
+    def enterCleanup(self):
+        """
+        Called when start using stacked cleanups.
+        """
+        self._cleanup_stack.append(self.__cleanup__)
+        self.__cleanup__ = []
+
+    def exitCleanup(self):
+        """
+        To be called at the end of a stacked cleanup.
```

Comment:
This works well, but could we forget to call `exitCleanup`? 

If you feel like it, perhaps add an assertion somewhere (but not sure where). Or perhaps a context manager.